src/Generator.cpp

```diff
@@ -21,362 +26,655 @@
-      pos(0)=x;
-      pos(1)=y;
-
-      if (debug) {
-        std::cout << " new simulated hit with energy " << denrj << " and position(x,y) " 
-          <<"("<<x<<","<<y<<")"<< " cell(i,j) " 
-          <<"("<<geometry_.closestCell(x,y).getIIndex()
-          <<","<<geometry_.closestCell(x,y).getJIndex()
-          <<")"<<" cell(x,y) "
-          <<"("<<geometry_.closestCell(x,y).getPosition()(0)
-          <<","<<geometry_.closestCell(x,y).getPosition()(1)
-          <<")"
-          <<" isincell(cell) "<<geometry_.isInCell(pos, geometry_.closestCell(x,y))
-          <<" position in cell " 
-          <<"("<<geometry_.positionInCell(pos)(0)
-          <<","<<geometry_.positionInCell(pos)(1)
-          <<")"
-          <<std::endl;
-      }
-
-      energygen += denrj;
-
-      // map generated point into geometry
-      const Cell& cell = geometry_.closestCell(x,y);
-
-      // for half-cell or boarder cells, check it is within the cell
-      bool isincell = geometry_.isInCell(pos, cell);
-      if (!isincell) { 
-        std::cout << "[main] point is not inside the closest cell!  x,y" << x << " " << y << 
-          " cell position " << cell.getPosition()(0) << " " << cell.getPosition()(1) << 
-          " closest cell indices " << cell.getIIndex() << " " <<  cell.getJIndex()<< std::endl;
-      }
-
-      // add energy to corresponding cell
-      if (isincell) {
-        event.fillHit(cell.id(), denrj);
-        energygenincells += denrj;  
-        energyrec += denrj; 
-      }	
-
-     // fill shower histograms
-      hTransverseProfile.Fill(r_shower,denrj);
-      hPhiProfile.Fill(phi_shower,denrj);
-      hSpotEnergy.Fill(denrj);
+    if (parameters_.general().debug)
+        geometry_.print();
 
-    }
+    cout << "Done !"<<endl;
 
-    std::cout << "simulated energy " << energygen << std::endl;
-    std::cout << "simulated energy inside cells " << energygenincells << std::endl;
-    std::cout << "reconstructed energy inside cells (includes noise) " << energyrec << std::endl;
-    
-    std::unique_ptr<ShowerShape> aShowerShape;
-    if (parameters_.geometry().type!=Parameters::Geometry::Type::Triangles) { // hexagons
-      aShowerShape.reset(new ShowerShapeHexagon(event.hits(), geometry_.getCells()));
-    } else { // triangles
-      aShowerShape.reset(new ShowerShapeTriangle(event.hits(), geometry_.getCells()));   
-    }  
-    std::cout << "cell max i,j " << aShowerShape->maxCell()->getIIndex() << " " << aShowerShape->maxCell()->getJIndex()
-    << " with energy " << aShowerShape->maxE1() << std::endl;
-    std::cout << "energy in first neighboors " << aShowerShape->firstNeighboors() << std::endl;
-
-    // fill histograms
-    hEnergyGen.Fill(energygen,1.);
-    hEnergySum.Fill(energyrec,1.);
-
-    for (const auto& id_energy : event.hits()) { 
-      hCellEnergyMap.at(id_energy.first).Fill(id_energy.second);
+    // Noise calibration of each cells for all layers
+    double** calibratednoise = new double*[52];
```

Comment:
Don't use `new`. This way it is automatically deleted.

Reply:
Fixed